service/matching/matchingEngine.go

```diff
@@ -236,6 +253,22 @@ func (e *matchingEngineImpl) getTaskQueueManager(taskQueue *taskQueueID, taskQue
 	return mgr, nil
 }
 
+// Returns taskQueueManager for a task queue. If not already cached gets new range from DB and
+// if successful creates one.
+// The taskQueueManager should be initialized (have acquired a lease and loaded metadata).
+func (e *matchingEngineImpl) getTaskQueueManagerInitialized(ctx context.Context, taskQueue *taskQueueID, taskQueueKind enumspb.TaskQueueKind) (taskQueueManager, error) {
+	mgr, err := e.getTaskQueueManager(taskQueue, taskQueueKind)
```

Comment:
Is this the only place to call getTaskQueueManager? Could the getTaskQueueManager just return the initialized TQMgr?

Reply:
Yeah, I do like it better. It's a little ugly in that function but better overall. I also combined it with IfAlreadyLoaded.